lumicks/pylake/population/dwelltime.py

```diff
@@ -145,37 +147,58 @@ def plot_axes(data, key, component, use_index):
         plt.tight_layout()
 
 
-@dataclass(frozen=True)
-class BindingDwelltimes:
-    """Results of exponential mixture model optimization for binding dwelltimes.
+class DwelltimeModel:
+    """Exponential mixture model optimization for dwelltime analysis.
 
-    This class is returned from `_kinetic_mle_optimize()` and should not be
-    constructed manually.
-
-    Attributes
+    Parameters
     ----------
+    dwelltimes : np.ndarray
+        observations on which the model was trained. *Note: the units of the optimized lifetime
+        will be in the units of the dwelltime data. If the dwelltimes are calculated as the number
+        of frames, this the needs to be multiplied by the frame time in order to obtain the
+        physically relevant parameter.*
     n_components : int
         number of components in the model.
-    dwelltimes_sec : np.ndarray
-        observations on which the model was trained.
-    observations_limits : tuple
-        tuple of (`min`, `max`) values of the experimental observation time.
-    _parameters : np.ndarray
-        optimized parameters in the order [amplitudes, lifetimes]
-    log_likelihood : float
-        log likelihood of the trained model
-    bootstrap : BindingDwelltimesBootstrap
-        object containing information about the bootstrapping analysis.
+    min_observation_time : float
+        minimum experimental observation time
+    max_observation_time : float
+        maximum experimental observation time.
+    tol : float
+        The tolerance for optimization convergence. This parameter is forwarded as the `ftol` argument
+        to `scipy.minimize(method="L-BFGS-B")`.
```

Comment:
Don't we use `SLSQP`?

Also, maybe just call  it `ftol` if that's the argument it is forwarded to. While I'm not a huge fan of their naming convention, it's one less mental translation that people have to make.

Reply:
oops...

I get what you mean about `ftol`, but I would argue it's more important to keep our syntax consistent rather than staying in line with the syntax used by the optimizers in the backend. This way, people using `pylake` only ever have to remember `tol` instead of trying to remember which of our functions use `tol` vs `ftol` etc... This is also why here I used our `max_iter` rather than the `scipy` `maxiter` arg

Reply:
Fair enough!